### PR TITLE
eth/tracers: use atomic type

### PR DESCRIPTION
**Pull request by s7v7nislands**

### CODE REVIEW

**Review by rjl493456442 — APPROVED** [review]

**Comment by holiman** [comment]
```
eth/tracers/api_test.go:892:67: copylocks: call of t.Errorf copies lock value: sync/atomic.Uint32 contains sync/atomic.noCopy (govet)
			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", ref, rel)
			                                                               ^
```

**Comment by holiman** [comment]
```
		if ref != rel {
			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", ref, rel)
		}
```
needs to be e.g
```
		if a ,b := re.Load(), rel.Load() {
			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", a,  b)
		}
```

**Comment by s7v7nislands** [comment]
I run make lint in locally, it will fail. But in CI it seems to be successful.
```
go-ethereum on  atomic10 via 🐳 orbstack via 🐹 v1.20.2 
❯ make lint
env GO111MODULE=on go run build/ci.go lint
build/cache/golangci-lint-1.51.1-darwin-amd64.tar.gz is up-to-date
>>> build/cache/golangci-lint-1.51.1-darwin-amd64/golangci-lint run --config .golangci.yml ./...
core/blockchain_test.go:3892:6: SA5011: possible nil pointer dereference (staticcheck)
                if *tail != expTail {
                   ^
core/blockchain_test.go:3889:6: SA5011(related information): this check suggests that the pointer can be nil (staticcheck)
                if tail == nil {
                   ^
util.go:48: exit status 1
exit status 1
make: *** [lint] Error 1
```

File: eth/tracers/api_test.go [hunk 1]
```diff
@@ -889,7 +889,7 @@ func TestTraceChain(t *testing.T) {
 			t.Error("Missing tracing block")
 		}
 		if ref != rel {
-			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", ref, rel)
+			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", ref.Load(), rel.Load())
```
Comment (holiman) [inline eth/tracers/api_test.go:892]: Myeah, I think it is not right to do `ref != rel` on two atomic types, on the line above. That's why I suggested using `have, want := ... ` instead

Comment (s7v7nislands) [reply eth/tracers/api_test.go]: my fault! update it.

File: eth/tracers/api_test.go [hunk 2]
```diff
@@ -888,8 +888,9 @@ func TestTraceChain(t *testing.T) {
 		if next != c.end+1 {
 			t.Error("Missing tracing block")
 		}
-		if ref != rel {
-			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", ref, rel)
+		loadRef, loadRel := ref.Load(), rel.Load()
+		if loadRef != loadRel {
+			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", loadRef, loadRel)
```
Comment (holiman) [inline eth/tracers/api_test.go:893]:
```suggestion
		
		if nref, nrel := ref.Load(), rel.Load(); nref != nrel {
			t.Errorf("Ref and deref actions are not equal, ref %d rel %d", nref, nrel)
```

**Review by holiman — APPROVED** [review]
LGTM

File: eth/tracers/native/prestate.go [hunk 3]
```diff
@@ -259,7 +259,7 @@ func (t *prestateTracer) GetResult() (json.RawMessage, error) {
 // Stop terminates execution of the tracer at the first opportune moment.
 func (t *prestateTracer) Stop(err error) {
 	t.reason = err
-	atomic.StoreUint32(&t.interrupt, 1)
+	t.interrupt.Store(true)
```
Comment (s1na) [inline eth/tracers/native/prestate.go:262]: I just realized through your PR that prestateTracer is ignoring the `interrupt` signal sort of. It has to check for interrupts in `CaptureState`. I'd be grateful if you add it in this PR, but don't feel obliged.

**Pull request merged**